### DISCUSSION TITLE
healthcheck 默认携带的host是不是带上port更合理一些？

### DISCUSSION THREAD

**Discussion by Lewisyixin** [apache/apisix#8572] [Ideas]
![image](https://user-images.githubusercontent.com/17898688/209607108-00ed4dac-af33-4050-8ce8-a28846f11070.png)
根据抓包结果查看，service/upstream 的主动健康检测默认携带的Host头只是ip，
根据http协议默认规范来看，Host头应该是ip:port(比如浏览器访问或者curl一个ip:port，默认都是携带ip:port作为host头)

**Comment by Lewisyixin** [comment]
看了下代码  应该是这个位置控制的，本地修改了下是生效的 不知道有没有其他问题
`deps/share/lua/5.1/resty/healthcheck.lua` line 848
`local request = ("GET %s HTTP/1.0\r\n%sHost: %s\r\n\r\n"):format(path, headers, hostheader or hostname or ip)`
改为
`local request = ("GET %s HTTP/1.0\r\n%sHost: %s\r\n\r\n"):format(path, headers, hostheader or hostname or ip..":"..port)`

**Reply by tokers** [reply]
@Lewisyixin Sounds reasonable, but I also want to know some scenarios that the default host without the port will cause some issues.

**Reply by Lewisyixin** [reply]
我们apisix上游有一个鉴权功能的网关，这个鉴权网关也是根据域名+url做得区分并且往他的上游转发，鉴权网关自身的健康检测接口为了避免和网关上游服务的健康检测冲突，所以规定了如果Host是默认值(也就是鉴权网关机器自身的ip:port)的时候，就当成鉴权网关自身的健康检测。
之前就发现apisix后面配置鉴权网关的时候 ，日志一直报错健康检测失败，今天抓包查了一下问题就在于apisix这边健康检测默认传的时候把端口去掉了。

**Reply by Lewisyixin** [reply]
虽然这种场景实际情况并不多，但是Http协议规范里Host就是包括主机名(ip)+端口的，所以感觉默认遵循这个规范也是应当的
![image](https://user-images.githubusercontent.com/17898688/209648403-02e23821-c9fd-4785-96f4-64e98b6e8a7a.png)

**Reply by tokers** [reply]
I agree with you. Let's see how others react.
